### PR TITLE
feat: make `ytt` pipeline test work

### PR DESCRIPTION
**Pull request by yihong0618**
this `PR` do these things:

1. add test to `cmake`
2. fix `rclient` work
3. make pipeline with `plcontainer` test work

will fix the whole tests in the future.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,4 +12,5 @@ src/common/config.h
 src/pyclient/bin/py3client
 build*/
 downloads*/
-
+# for test
+tests/regress/results
```
Comment (Sasasu) [inline .gitignore:16]: add a new line here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -169,19 +169,23 @@ install(
...
     PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/rclient/librcall.so
-    DESTINATION bin/plcontainer_clients/)
+    DESTINATION bin/plcontainer_clients/ OPTIONAL)
 install(
     PROGRAMS src/rclient/bin/rclient.sh
-    DESTINATION bin/plcontainer_clients/)
+    DESTINATION bin/plcontainer_clients/ OPTIONAL)
+
+# Add tests targets
+add_subdirectory(tests)
```
Comment (Sasasu) [inline CMakeLists.txt:191]: and here

**Review by Sasasu — COMMENTED** [review]
not check the test case yet.
looks the python3 test input and output are copied from python2 test, ok to me

**Review by Sasasu — APPROVED** [review]

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    SQL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/isolation2/sql
+    EXPECTED_DIR ${CMAKE_CURRENT_SOURCE_DIR}/isolation2/expected
+    RESULTS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/isolation2/results
+    DATA_DIR ${CMAKE_CURRENT_SOURCE_DIR}/data
+    SCHEDULE_FILE ${CMAKE_CURRENT_SOURCE_DIR}/isolation2/isolation2_schedule
+    REGRESS_OPTS
+    --dbname=isolation2test)
+
+add_custom_target(installcheck)
+add_dependencies(installcheck testpy39 testr)
```
Comment (beeender) [inline tests/CMakeLists.txt:50]: do we want to add `isolation2` as well?

File: concourse/scripts/test_plcontainer_py39.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -l
+
+set -exo pipefail
+
+function _main() {
+    # Start gpdb cluster
+    export MASTER_DATA_DIRECTORY=/data/gpdata/master/gpseg-1
+    source /usr/local/greenplum-db-devel/greenplum_path.sh
+    source gpdb_src/gpAux/gpdemo/gpdemo-env.sh
```
Comment (beeender) [inline concourse/scripts/test_plcontainer_py39.sh:9]: I don't think these 3 lines are needed. They have been added to `.bashrc` already. see https://github.com/greenplum-db/plcontainer/pull/640/files#diff-b3f677ed5ed27ac32eb32afb3ded43d98990d32881dce0055e15bc991ffe44e9R181

File: concourse/scripts/test_plcontainer_r.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash -l
+
+set -exo pipefail
+
+function _main() {
+    # Start gpdb cluster
+    export MASTER_DATA_DIRECTORY=/data/gpdata/master/gpseg-1
+    source /usr/local/greenplum-db-devel/greenplum_path.sh
+    source gpdb_src/gpAux/gpdemo/gpdemo-env.sh
```
Comment (beeender) [inline concourse/scripts/test_plcontainer_r.sh:9]: ditto

File: concourse/scripts/test_plcontainer_py39.sh (hunk 4 above)
Comment (yihong0618) [reply concourse/scripts/test_plcontainer_py39.sh:9]: copy that.

**Review by beeender — APPROVED** [review]

**Pull request merged**